kolibri/plugins/facility/assets/src/composables/useUserManagement.js

```diff
@@ -0,0 +1,85 @@
+import { ref, computed, getCurrentInstance, watch } from 'vue';
+import pickBy from 'lodash/pickBy';
+import FacilityUserResource from 'kolibri-common/apiResources/FacilityUserResource';
+import ClassroomResource from 'kolibri-common/apiResources/ClassroomResource';
+import { _userState } from '../modules/mappers';
+
+export default function useUserManagement(activeFacilityId) {
+  const facilityUsers = ref([]);
+  const totalPages = ref(0);
+  const usersCount = ref(0);
+  const dataLoading = ref(false);
+  const classes = ref([]);
+  const store = getCurrentInstance().proxy.$store;
+  const route = computed(() => store.state.route);
+  // query params
+  const page = computed(() => Number(route.value.query.page) || 1);
+  const pageSize = computed(() => Number(route.value.query.page_size) || 30);
+  const ordering = computed(() => route.value.query.ordering || null);
+  const order = computed(() => route.value.query.order || '');
+  const search = computed(() => route.value.query.search || null);
+  const userType = computed(() => route.value.query.user_type || null);
+
+  const fetchUsers = async () => {
+    dataLoading.value = true;
+    try {
+      const resp = await FacilityUserResource.fetchCollection({
+        getParams: pickBy({
+          member_of: activeFacilityId,
+          page: page.value,
+          page_size: pageSize.value,
+          search: search.value?.trim() || null,
+          ordering: order.value === 'desc' ? `-${ordering.value}` : ordering.value || null,
+          user_type: userType.value,
+        }),
+        force: true,
```

Comment:
I see that we have `force: true` is there a specific reason we are doing this? Trying to think this out loud, so if User clicks "Sort by UserName" -> we do an API call, if the User clicks again the same sorting field then this would do the API call again right? is there a way we can prevent this?

Reply:
Yes, I updated the `changeSortHandler()` func to see if the same sorting settings are already being used in the current query params, to prevent duplicate API requests if they haven’t changed.